src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -185,6 +189,40 @@ private function extractFromAccessor(string $class, string $property): ?array
         return null;
     }
 
+    /**
+     * Tries to extract type information from constructor.
+     *
+     * @return Type[]|null
+     */
+    private function extractFromConstructor(string $class, string $property): ?array
+    {
+        try {
+            $reflectionClass = new \ReflectionClass($class);
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+
+        $constructor = $reflectionClass->getConstructor();
+
+        if (!$constructor) {
+            return null;
+        }
+
+        foreach ($constructor->getParameters() as $parameter) {
+            if ($property !== $parameter->name) {
+                continue;
+            }
+
+            return array($this->extractFromReflectionType($parameter->getType()));
+        }
+
+        if ($reflectionClass->getParentClass()) {
```

Comment:
`$parent =` to prevent creating two instances of ReflectionClass

Reply:
done.